lib/phoenix_live_view/html_engine.ex

```diff
@@ -1040,23 +1107,84 @@ defmodule Phoenix.LiveView.HTMLEngine do
     end
   end
 
-  defp store_component_call(module, component, attrs, file, line) do
+  defp store_component_call(module, component, slots, attrs, file, line, env) do
     if Module.open?(module) do
+      pruned_slots =
+        for {slot_name, slot_values} <- slots,
+            do: {slot_name, Enum.map(slot_values, &slot_call_value(&1, env))},
+            into: %{}
+
       pruned_attrs =
         for {attr, value, meta} <- attrs,
             is_binary(attr) and not String.starts_with?(attr, ":"),
-            do: {String.to_atom(attr), {meta[:line], meta[:column], component_call_value(value)}},
+            {kind, value} = component_call_value(value, env),
+            do: {String.to_atom(attr), {meta[:line], meta[:column], kind, value}},
             into: %{}
 
       root = List.keymember?(attrs, :root, 0)
-      call = %{component: component, attrs: pruned_attrs, file: file, line: line, root: root}
+
+      call = %{
+        component: component,
+        slots: pruned_slots,
+        attrs: pruned_attrs,
+        file: file,
+        line: line,
+        root: root
+      }
+
       Module.put_attribute(module, :__components_calls__, call)
     end
   end
 
-  defp component_call_value({:expr, _, _}), do: :expr
-  defp component_call_value({:string, string, _}), do: string
-  defp component_call_value(nil), do: nil
+  defp component_call_value({:string, value, _meta}, _env) do
+    {:lit, value}
+  end
+
+  defp component_call_value(nil, _env) do
+    {:lit, true}
+  end
+
+  defp component_call_value({:expr, value, %{line: line, column: column}}, env) do
+    case Code.string_to_quoted!(value, line: line, column: column, file: env.file) do
```

Comment:
We already call `string_to_quoted` in `parse_expr!`. We should avoid calling it twice because if there is any warning in the code, the warning will be emitted twice. We should reuse the result from parse_expr!.